### PR TITLE
feat: created TypeScript sample app

### PR DESCRIPTION
**Pull request by btw17**
Created a mirror of `./examples/sample-app` converted to TypeScript at `./examples/sample-app-ts`

## Notes

- In it's current state, this app is translated to TS and is fully operational in the same ways the original sample-app is, though it's done nothing to update the code conventions or add additional type information beyond what is available in in the `blockly` core package.
- Additional considerations:
  - Updating the `css` implementation to `scss`

### CODE REVIEW

File: examples/sample-app-ts/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "sample-app",
```
Comment (maribethb) [inline examples/sample-app-ts/package.json]: can you make this `sample-app-typescript`, we use the package names for some things and I think there is a potential for breakage if we have duplicates

File: examples/sample-app-ts/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        "declarationMap": true,
+        "module": "ES2015",
+        "moduleResolution": "node",
+        "target": "ES2015",
+        "strict": true,
+    },
+    // NOTE: `test/**/*` is automatically included in `blockly-scripts start`.
+    // Only src matters for production builds.
+    "exclude": ["test/**/*"]
+}
```
Comment (maribethb) [inline examples/sample-app-ts/tsconfig.json]: needs newline at eof here

File: examples/sample-app-ts/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        "baseUrl": "./",
+        "outDir": "dist",
+        "declaration": true,
+        "declarationMap": true,
+        "module": "ES2015",
+        "moduleResolution": "node",
+        "target": "ES2015",
+        "strict": true,
+    },
+    // NOTE: `test/**/*` is automatically included in `blockly-scripts start`.
```
Comment (maribethb) [inline examples/sample-app-ts/tsconfig.json]: The comment is definitely wrong (we aren't using `blockly-scripts start`), but I think this entire property needs to go as currently there is no `test` directory. Eventually it should have tests, but if the tests are in ts too then i'm not sure if we would want this or not.

Comment (btw17) [reply examples/sample-app-ts/tsconfig.json]: Good catch! Fixed.

File: examples/sample-app-ts/tsconfig.json (hunk 2 above)
Comment (btw17) [reply examples/sample-app-ts/tsconfig.json]: Done!

File: examples/sample-app-ts/package.json (hunk 1 above)
Comment (btw17) [reply examples/sample-app-ts/package.json]: Done!

**Review by maribethb — APPROVED** [review]

**Pull request merged**